lib/private/OCS/ApiHelper.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * SPDX-FileCopyrightText: 2016-2024 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-FileCopyrightText: 2016 ownCloud, Inc.
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+
+namespace OC\OCS;
+
+use OC\AppFramework\OCS\V1Response;
+use OC\AppFramework\OCS\V2Response;
+use OCP\AppFramework\Http\DataResponse;
+use OCP\AppFramework\OCSController;
+use OCP\IRequest;
+use OCP\Server;
+
+class ApiHelper {
+	/**
+	 * Respond to a call
+	 * @psalm-taint-escape html
+	 * @param int $httpStatusCode force the HTTP status code, only used for the special case of maintenance mode which return 503 even for v1
+	 */
+	public static function respond(int $statusCode, string $statusMessage, array $headers = [], ?int $httpStatusCode = null): void {
```

Comment:
Maybe more explicit?

```suggestion
	public static function respond(int $statusCode, string $statusMessage, array $headers = [], ?int $overrideStatusCode = null): void {
```